core/controllers/blog_homepage.py

```diff
@@ -49,9 +49,18 @@ def _get_blog_card_summary_dicts_for_homepage(summaries):
     for summary in summaries:
         summary_dict = summary.to_dict()
         user_settings = user_services.get_user_settings(
+            summary_dict['author_id'], strict=False)
+        if user_settings:
+            summary_dict['profile_pic_url'] = (
+                user_settings.profile_picture_data_url)
+            summary_dict['author_username'] = user_settings.username
+        else:
+            summary_dict['profile_pic_url'] = None
+            summary_dict['author_username'] = 'author account deleted'
+        author_details = blog_services.get_blog_author_details(
```

Comment:
same comment as above.

Reply:
Just giving an idea. If you are sure that `get_blog_author_details` can never return None then raise an error in the `get_blog_author_details` function itself if the None value is encountered.

Reply:
Same as above